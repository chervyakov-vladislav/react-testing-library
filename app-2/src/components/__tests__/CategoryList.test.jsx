import { screen } from '@testing-library/react';

import { renderWithRouter } from '../../testUtils/routerProvider';
import { CategoryList } from '../CategoryList';

describe('CategoryList', () => { 
  it('рендер компонента без содержимого', () => {
    renderWithRouter(<CategoryList />);

    expect(screen.getByRole('list')).toMatchSnapshot();
  });

  it('рендер компонента с дочерними элементами', () => {
    renderWithRouter(<CategoryList catalog={[
      {
        idCategory: 1,
        strCategory: 'sadasdasd',
        strCategoryThumb: 'adasdasd',
        strCategoryDescription: 'asdadsasd'
      },
      {
        idCategory: 2,
        strCategory: 'asdasdasd',
        strCategoryThumb: 'asdasda',
        strCategoryDescription: 'asdasdasd'
      }
    ]} />);

    expect(screen.getByRole('list')).toMatchSnapshot();
  });
});