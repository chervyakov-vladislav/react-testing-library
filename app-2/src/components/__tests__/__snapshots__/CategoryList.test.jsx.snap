// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList рендер компонента без содержимого 1`] = `
<div
  class="list"
  role="list"
/>
`;

exports[`CategoryList рендер компонента с дочерними элементами 1`] = `
<div
  class="list"
  role="list"
>
  <div
    class="card"
    role="article"
  >
    <div
      class="card-image"
    >
      <img
        alt="sadasdasd"
        src="adasdasd"
      />
    </div>
    <div
      class="card-content"
    >
      <span
        class="card-title"
      >
        sadasdasd
      </span>
      <p>
        asdadsasd
        ...
      </p>
    </div>
    <div
      class="card-action"
    >
      <a
        class="btn"
        href="/category/sadasdasd"
      >
        Watch category
      </a>
    </div>
  </div>
  <div
    class="card"
    role="article"
  >
    <div
      class="card-image"
    >
      <img
        alt="asdasdasd"
        src="asdasda"
      />
    </div>
    <div
      class="card-content"
    >
      <span
        class="card-title"
      >
        asdasdasd
      </span>
      <p>
        asdasdasd
        ...
      </p>
    </div>
    <div
      class="card-action"
    >
      <a
        class="btn"
        href="/category/asdasdasd"
      >
        Watch category
      </a>
    </div>
  </div>
</div>
`;
